engine/runtime/src/main/java/org/enso/interpreter/node/callable/InvokeMethodNode.java

```diff
@@ -274,6 +298,17 @@ Object doFunctionalDispatchUncachedSymbol(
     return invokeFunctionNode.execute(function, frame, state, arguments);
   }
 
+  private static boolean isNamedAsAssociatedType(Type t) {
+    var at = t.getDefinitionScope().getAssociatedType();
+
+    // XXX doesn't work for Polyglot - it is there 3x times!
+    // return at == t;
```

Comment:
I don't think I understand what this comment is trying to tell.

Reply:
What is the associated type and why does it need special handling?

Reply:
From Enso perspective, there is no difference between a *module* and a *type*. On both of these, you can execute methods. An `associatedType` is an internal engine way how to distinguish these cases. Every module, has an `associatedType`, which represents that module as a type. For example:
`My_Module.enso`
```
type My_Type
```
An associatedType for `My_Module` module would be a type with `My_Type` members.

Reply:
AH, thanks for the explanation.

Reply:
Although btw. while not very relevant to this PR, it is not _entirely_ true that there is no difference. Types can be used in signatures, modules cannot. I just ran into it on my current task a few minutes ago:
```
error: The name AWS_Region resolved to a module, but modules are not allowed in type signature.
   36 |     With_Advanced_Properties (base_credential : AWS_Credential) (default_region : AWS_Region)
```

(I messed up an import and imported just the parent module, not the type define inside of it.)

Reply:
That is probably a bug in type ascriptions. Modules should be allowed for type ascriptions as well. No?

Reply:
It does not look like a bug given that the error message seems to be carefully written - more like a feature someone consciously implemented :)

I don't think we have a use-case for needing a module in a type ascription yet, so for now that does not seem like an issue.

Reply:
> Modules should be allowed for type ascriptions as well. No?

Not right now. When @kustosz implemented proper type signature checking
- #3511

the biggest set of errors was related to the fact that we name _module and its type the same_. Many signature were just wrong because they referred to module, but wanted to refer to a type. As such Marcin made it an error.

I believe we can keep it this way for a foreseeable future.